contributors/design-proposals/device-plugin.md

```diff
@@ -0,0 +1,503 @@
+Device Manager Proposal
+===============
+
+* [Motivation](#motivation)
+* [Use Cases](#use-cases)
+* [Objectives](#objectives)
+* [Non Objectives](#non-objectives)
+* [Vendor story](#vendor-story)
+* [End User story](#end-user-story)
+* [Device Plugin](#device-plugin)
+    * [Introduction](#introduction)
+    * [Registration](#registration)
+    * [Unix Socket](#unix-socket)
+    * [Protocol Overview](#protocol-overview)
+    * [API specification](#api-specification)
+    * [HealthCheck and Failure Recovery](#healthcheck-and-failure-recovery)
+    * [API Changes](#api-changes)
+* [Upgrading your cluster](#upgrading-your-cluster)
+* [Installation](#installation)
+* [Versioning](#versioning)
+* [References](#references)
+
+_Authors:_
+
+* @RenaudWasTaken - Renaud Gaubert &lt;rgaubert@NVIDIA.com&gt;
+* @jiayingz - Jiaying Zhang &lt;jiayingz@google.com&gt;
+
+# Motivation
+
+Kubernetes currently supports discovery of CPU and Memory primarily to a
+minimal extent. Very few devices are handled natively by Kubelet.
+
+It is not a sustainable solution to expect every hardware vendor to add their
+vendor specific code inside Kubernetes to make their devices usable.
+
+Instead, we want a solution for vendors to be able to advertise their resources
+to Kubelet and monitor them without writing custom Kubernetes code.
+We also want to provide a consistent and portable solution for users to
+consume hardware devices across k8s clusters.
+
+This document describes a vendor independant solution to:
+  * Discovering and representing external devices
+  * Making these devices available to the containers, using these devices,
+    scrubbing and securly sharing these devices.
+  * Health Check of these devices
+
+Because devices are vendor dependant and have their own sets of problems
+and mechanisms, the solution we describe is a plugin mechanism that may run
+in a container deployed through the DaemonSets mechanism.
```

Comment:
Just to raise for the sake of alignment if possible: The delivery and a (little bit) discovery mechanism look similar to the one used here kubernetes/community#589